sycl/include/CL/sycl/detail/boolean.hpp

```diff
@@ -56,11 +56,11 @@ template <int N> struct Boolean {
   using DataType = element_type[N];
 #endif
 
-  Boolean() : value{false} {}
+  Boolean() : value{0} {}
 
   Boolean(std::initializer_list<element_type> l) {
     for (size_t I = 0; I < N; ++I) {
-      value[I] = *(l.begin() + I);
+      value[I] = (!!*(l.begin() + I)) * (-1);
```

Comment:
It's hard to understand what is happening here (probably I still don't, so suggestion is not correct). Can we rewrite it to make it dummy?
```suggestion
      value[I] = (*(l.begin() + I))? -1 : 0;
```